Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -0,0 +1,823 @@
+        var parent = tile.parent;
+        if (!defined(parent)) {
+            return isVisible(tile.visibilityPlaneMask);
+        }
+        var showAdditive = parent.refine === Cesium3DTileRefine.ADD && parent._screenSpaceError > maximumScreenSpaceError;
+
+        return isVisible(tile.visibilityPlaneMask) && (!showAdditive || getScreenSpaceError(this.tileset, parent.geometricError, tile, this.frameState) > maximumScreenSpaceError);
+    };
+
+    InternalSkipTraversal.prototype.leafHandler = function(tile) {
+        if (tile !== this.root) {
+            if (tile.refine === Cesium3DTileRefine.REPLACE && !childrenAreVisible(tile)) {
+                ++this.tileset._statistics.numberOfTilesCulledWithChildrenUnion;
+                return;
+            }
+            if (!tile.hasEmptyContent) {
+                if (this.tileset.loadSiblings) {
+                    var parent = tile.parent;
+                    var tiles = parent.children;
+                    var length = tiles.length;
+                    for (var i = 0; i < length; ++i) {
+                        loadTile(tiles[i], this.frameState);
+                        touch(this.tileset, tiles[i], this.outOfCore);
+                    }
+                } else {
+                    loadTile(tile, this.frameState);
+                    touch(this.tileset, tile, this.outOfCore);
+                }
+            }
+            this.queue.push(tile);
+        } else if (!hasAdditiveContent(tile)) {
+            // additive tiles have already been pushed
+            this.tileset._desiredTiles.push(tile);
+        }
+    };
+
+    function updateChildren(tileset, tile, frameState) {
+        if (tile._lastVisitedFrame === frameState.frameNumber) {
+            return tile.childrenVisibility;
+        }
+
+        var children = tile.children;
+
+        updateTransforms(children, tile.computedTransform);
+        computeDistanceToCamera(children, frameState);
+
+        return computeChildrenVisibility(tile, frameState);
+    }
+
+    function visitTile(tileset, tile, frameState, outOfCore) {
+        // because the leaves of one tree traversal are the root of the subsequent traversal, avoid double visitation
+        ++tileset._statistics.visited;
+        tile.selected = false;
+        tile._finalResolution = false;
+        computeSSE(tile, frameState);
+        touch(tileset, tile, outOfCore);
+        tile._ancestorWithContent = undefined;
+        tile._ancestorWithLoadedContent = undefined;
+        var parent = tile.parent;
+        if (defined(parent)) {
+            var replace = parent.refine === Cesium3DTileRefine.REPLACE;
+            tile._ancestorWithContent = (replace && parent.hasRenderableContent) ? parent : parent._ancestorWithContent;
+            tile._ancestorWithLoadedContent = (replace && parent.hasRenderableContent && parent.contentReady) ? parent : parent._ancestorWithLoadedContent;
+        }
+    }
+
+    function touch(tileset, tile, outOfCore) {
```

Comment:
Is it worth adding this? The other flags are important because they prevent tiles from getting put onto a list twice, but this just moves the tile in the doubly linked list.